src/components/Reports/ReportsBox/ReportsBox.test.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { render, cleanup } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import DopplerIntlProvider from '../../../i18n/DopplerIntlProvider.double-with-ids-as-values';
+import ReportsBox from './ReportsBox';
+import { addDays } from '../../../utils';
+
+describe('ReportsBox component', () => {
+  afterEach(cleanup);
+
+  it('should show loading when visits are null or undefined', () => {
```

Comment:
Is this ok? when there are no visits the spinner would indicate that this is loading forever ...

Reply:
Good catch @cbernat!

Reply:
its only when is null or undefined. when have 0 visits shows the box

Reply:
What do you think about making the loading state explicit with a flag? (in general, it is better being explicit)

For that cases, using used to be better, you can limit the possible outputs.

Reply:
yes i already did now

Reply:
I've talk this with @fcoronelmakingsense and it seems when the promise is being resolved we have a temporal value of null or undefined, while waiting for the request, perhaps it would be great to have a flag for loading, in order to differentiate, null/undefined in loading and null/undefined on error.